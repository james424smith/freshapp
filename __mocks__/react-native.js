import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
