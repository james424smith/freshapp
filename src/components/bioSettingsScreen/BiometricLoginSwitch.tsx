import React from "react";
import { View, Switch } from "react-native";
import styles from "../../styles/more_screens/BiometricLoginStyle";
import colors from "../../constants/styles/colors";
import Text from "../../components/StyledText";
import { Icon } from "react-native-elements";
import { useTheme } from "@react-navigation/native";

type Props = {
  isEnabled: boolean;
  toggleSwitch: any;
};

const BiometricLoginSwitch = (props: Props) => {
  const theme = useTheme();
  const { isEnabled, toggleSwitch } = props;
  return (
    <View style={styles.mainStyle}>
      <Icon
        name="fingerprint"
        size={25}
        color={colors.white}
        style={styles.iconStyle}
      />
      <View style={styles.content}>
        <Text style={{ ...styles.text, color: theme.colors.text }}>
          Enable Biometrics
        </Text>
        <Switch
          trackColor={{ false: colors.inactiveGrey, true: colors.marlowBlue }}
          thumbColor={isEnabled ? colors.white : colors.grey}
          onValueChange={toggleSwitch}
          value={isEnabled}
        />
      </View>
    </View>
  );
};
export default BiometricLoginSwitch;
