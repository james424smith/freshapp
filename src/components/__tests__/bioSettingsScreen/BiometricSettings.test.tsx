import React from "react";
import { render, cleanup } from "@testing-library/react-native";

import BiometricLoginSwitch from "../../bioSettingsScreen/BiometricLoginSwitch";

describe("SignInScreen test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correctly as per the ui made when the toggle is enabled ", () => {
    const wrapper = render(
      <BiometricLoginSwitch toggleSwitch={true} isEnabled={true} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it("should render correctly as per the ui made when the toggle is disabled", () => {
    const wrapper = render(
      <BiometricLoginSwitch toggleSwitch={false} isEnabled={false} />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
