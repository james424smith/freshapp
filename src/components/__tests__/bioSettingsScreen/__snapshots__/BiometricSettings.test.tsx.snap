// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInScreen test should render correctly as per the ui made when the toggle is disabled 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flexDirection": "row",
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#78849E",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 25,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#C5C5C5",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Enable Biometrics
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#005AA5"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      thumbTintColor="#78849E"
      tintColor="#C5C5C5"
      value={false}
    />
  </View>
</View>
`;

exports[`SignInScreen test should render correctly as per the ui made when the toggle is enabled  1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flexDirection": "row",
      "paddingBottom": 15,
      "paddingTop": 15,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#78849E",
          "borderRadius": 20,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 25,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 1,
        "borderColor": "#C5C5C5",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginLeft": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#2A2E43",
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "color": "rgb(28, 28, 30)",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
    >
      Enable Biometrics
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#005AA5"
      style={
        Object {
          "height": 31,
          "width": 51,
        }
      }
      thumbTintColor="#FFFFFF"
      tintColor="#C5C5C5"
      value={true}
    />
  </View>
</View>
`;
